packages/contracts/scripts/smt-checker/README.md

```diff
@@ -0,0 +1,62 @@
+# SMT-Checker Support for Smart Contract Testing
+
+This repository provides SMT-Checker support for testing smart contracts. Follow these steps to get started:
+
+
+## Make Sure Your System Has Z3 Installed >= "4.8.11"
+
+Ensure that your system has Z3 installed with a version number greater than or equal to the required version.
+
+## Step 1: Download and Install Z3
+
+Before proceeding, ensure that you have `g++` and `python3` installed on your system. To download and install Z3, enter the following commands in your terminal:
+
```

Comment:
I would rather have a check inside the shell script to ensure the dependencies are there and if not print the link to the project i.e of an eye message " Z3 is require please check https://GitHub. Com/Z3prover for the installation procedure 
Indeed apt-get is not a command that you can run on every os

Reply:
You will need to absolutely install it using the following instructions, otherwise you will think you have "Z3" ubuntu ships Z3, but it's not enough, you need to pass the dynamic library (compiled) to the folder at  x86_64-linux-gnu. Otherwise the compiler (solidity) won't catch any Z3

Reply:
Yes the idea is to create a script that will check that and if it is installed already then will move forward and execute the other commands you specified in the doc

Reply:
It is never installed, the thing it's the library needs to be exactly **libz3.so.4.11**, that's why the script does `sudo mv libz3.so libz3.so.4.11` - please note the versioning at the name the library compiles and comes with ubuntu as libz3.so, and on ubuntu it does not comes with the header, with need everything in the steps to perform accordingly, i debugged this one thoroughly hard.

Regarding the apt-get, we're supporting Linux - Ubuntu on this PR

Reply:
pushed a script that does everything https://github.com/ubiquity/ubiquity-dollar/blob/aa2109926197aed781042491de602f47532b6db3/packages/contracts/scripts/smt-checker/update-dependencies.sh at https://github.com/ubiquity/ubiquity-dollar/pull/702/commits/aa2109926197aed781042491de602f47532b6db3

Reply:
Can you update the script and remove the apt-get and replace them by something like this
```
#!/bin/bash

# Check if wget is installed
if ! command -v wget &>/dev/null; then
    echo "The wget program is not installed on your system."
    echo "Please install it to continue."
    echo "On Ubuntu/Debian, use the command:"
    echo "sudo apt-get install wget"
    echo "On Arch Linux, use the command:"
    echo "sudo pacman -S wget"
    echo "On macOS, you can use Homebrew to install wget:"
    echo "brew install wget"
    exit 1
fi

# Check if g++ is installed
if ! command -v g++ &>/dev/null; then
    echo "The g++ compiler is not installed on your system."
    echo "Please install it to continue."
    echo "On Ubuntu/Debian, use the command:"
    echo "sudo apt-get install g++"
    echo "On Arch Linux, use the command:"
    echo "sudo pacman -S gcc"
    echo "On macOS, you can install g++ by installing Xcode Command Line Tools:"
    echo "xcode-select --install"
    exit 1
fi

echo "All required programs are installed on your system."
```

Reply:
I couldn't confirm that the path to **mv** the library is the same as linux **/usr/lib/x86_64-linux-gnu**, I think that's good for new github issues to support other platforms on this module!

Reply:
let's push this one, and assign me on a github issue to cover other platforms on this one! @zgorizzo69, yes including Windows

Reply:
@zgorizzo69  https://github.com/ubiquity/ubiquity-dollar/pull/702/commits/2a1b1f2bc00fb79dd3249e34c85566b729c077eb

Reply:
> I would rather have a check inside the shell script to ensure the dependencies are there and if not print the link to the project i.e of an eye message " Z3 is require please check https://GitHub. Com/Z3prover for the installation procedure Indeed apt-get is not a command that you can run on every os

the Z3 project won't ever tell you the installation requirement for the specific solidity compiler, the compiler needs the file to be **libz3.so.4.11** at **/usr/lib/x86_64-linux-gnu**